keras/callbacks.py

```diff
@@ -1817,6 +1822,12 @@ def on_epoch_end(self, epoch, logs=None):
     current = self.get_monitor_value(logs)
     if current is None:
       return
+    if self.mode == "min" and self.threshold != None and current < self.threshold:
+      self.model.stop_training = True
+      return
```

Comment:
You don't have to return from the function. If the current epoch was best and `restore_best_weights= True` then the weights wont ever be restored which is done in the code following this line.

Reply:
Done